frappe/boot.py

```diff
@@ -129,43 +129,68 @@ def get_user_pages_or_reports(parent, cache=False):
 
 	roles = frappe.get_roles()
 	has_role = {}
-	column = get_column(parent)
+
+	page_doctype = frappe.qb.DocType("Page")
+	report_doctype = frappe.qb.DocType("Report")
+
+	columns = (page_doctype.title.as_("title"),)
+	if parent == "Report":
+		columns = (
+			report_doctype.name.as_("title"),
+			report_doctype.ref_doctype,
+			report_doctype.report_type
+			)
+
+	custom_role_doctype = frappe.qb.DocType("Custom Role")
+	has_role_doctype = frappe.qb.DocType("Has Role")
+	parent_doctype = frappe.qb.DocType(parent)
 
 	# get pages or reports set on custom role
-	pages_with_custom_roles = frappe.db.sql("""
-		select
-			`tabCustom Role`.{field} as name,
-			`tabCustom Role`.modified,
-			`tabCustom Role`.ref_doctype,
-			{column}
-		from `tabCustom Role`, `tabHas Role`, `tab{parent}`
-		where
-			`tabHas Role`.parent = `tabCustom Role`.name
-			and `tab{parent}`.name = `tabCustom Role`.{field}
-			and `tabCustom Role`.{field} is not null
-			and `tabHas Role`.role in ({roles})
-	""".format(field=parent.lower(), parent=parent, column=column,
-		roles = ', '.join(['%s']*len(roles))), roles, as_dict=1)
+	pages_with_custom_roles = (
+		frappe.qb
+		.from_(custom_role_doctype)
+		.from_(has_role_doctype)
+		.from_(parent_doctype)
+		.select(
+			custom_role_doctype.field(parent.lower()).as_("name"),
```

Comment:
Was styling this query and this happened when I tried running it:

```python
TypeError                                 Traceback (most recent call last)
~/Desktop/frappe-bench-dev/apps/frappe/frappe/commands/utils.py in <module>
      5                 .from_(ParentTable)
      6 		.select(
----> 7                         CustomRole.field(parent.lower()).as_("name"),
      8                         CustomRole.modified,
      9                         CustomRole.ref_doctype,

TypeError: 'PseudoColumn' object is not callable
```

Are you sure this works?

Reply:
field doesn't work, replaced it with `customRole[parent.lower()]`